cpp/src/knn/knn_opg_common.cuh

```diff
@@ -27,6 +27,8 @@
 #include <raft/core/comms.hpp>
 #include <raft/core/handle.hpp>
 #include <raft/spatial/knn/knn.cuh>
+#include <raft/spatial/knn/specializations.cuh>
```

Comment:
These should be protected by the conditionals (RAFT_NN_COMPILED for spatial/knn/specializations and RAFT_DISTANCE_COMPILED for raft/distance/specializations)